app/models/pglogical_subscription.rb

```diff
@@ -0,0 +1,119 @@
+# This model then exposes select values returned from that method
+class PglogicalSubscription < ActsAsArModel
+  set_columns_hash(
+    :id              => :string,
+    :status          => :string,
+    :dbname          => :string,
+    :host            => :string,
+    :user            => :string,
+    :password        => :string,
+    :port            => :integer,
+    :provider_region => :integer
+  )
+
+  def self.find(*args)
+    subs = pglogical.enabled? ? pglogical.subscriptions : []
+    case args.first
+    when :all   then subs.collect { |s| new(subscription_to_columns(s)) }
+    when :first then subs.empty? ? nil : new(subscription_to_columns(subs.first))
+    when :last  then subs.empty? ? nil : new(subscription_to_columns(subs.last))
+    end
+  end
+
+  def save
+    save!
+    true
+  rescue
+    false
+  end
+
+  def save!
+    create_node.check if !pglogical.enabled? || self.class.count == 0
+    id = "subscription_#{host.gsub(/\.|-/, "_")}"
+    pglogical.subscription_create(id, dsn, ['miq'], false).check
+  end
+
+  def delete
+    pglogical.subscription_drop(id, true)
+    MiqRegion.destroy_region(connection, provider_region)
+    pglogical.node_drop(node_name, true) if self.class.count == 0
+  end
+
+  def self.delete_all
+    find(:all).each(&:delete)
+  end
+
+  def self.pglogical(refresh = false)
+    @pglogical = nil if refresh
+    @pglogical ||= connection.pglogical
+  end
+
+  def pglogical(refresh = false)
+    self.class.pglogical(refresh)
+  end
+
+  # translate the output from the pglogical stored proc to our object columns
+  def self.subscription_to_columns(sub)
+    cols = sub.symbolize_keys
+
+    # delete the things we dont care about
+    cols.delete(:slot_name)
+    cols.delete(:replication_sets)
+    cols.delete(:forward_origins)
+
+    cols[:id] = cols.delete(:subscription_name)
+
+    # create the individual dsn columns
+    cols.merge!(dsn_to_hash(cols.delete(:provider_dsn)))
+
+    cols[:provider_region] = cols.delete(:provider_node).sub("region_", "").to_i
+    cols
+  end
+  private_class_method :subscription_to_columns
+
+  def self.dsn_to_hash(dsn)
+    dsn.split.each_with_object({}) do |part, cols|
+      key, val = part.split("=")
+
+      # undo PG::Connection.quote_connstr
+      val.gsub!(/^'|'$/, "")
+      val.gsub!(/\\'/, "'")
+
+      case key
+      when "dbname", "host", "user" then cols[key.to_sym] = val
+      when "port" then cols[key.to_sym] = val.empty? ? nil : val.to_i
+      end
+    end
+  end
+  private_class_method :dsn_to_hash
+
+  private
+
+  def node_name
+    "region_#{MiqRegion.my_region_number}"
+  end
+
+  def create_node
+    pglogical.enable
+
+    node_dsn = PG::Connection.parse_connect_args(connection.raw_connection.conninfo_hash.delete_blanks)
+    pglogical.node_create(node_name, node_dsn)
+  end
+
+  def dsn
+    decrypted = MiqPassword.encrypted?(password) ? MiqPassword.decrypt(password) : password
```

Comment:
I need it for the dsn so there needs to be a way for the user of the model to set it before hitting save.
So I think I still need the attribute, but don't want it exposed afterwards.